### PR TITLE
beancount-language-server: refactor -> 1.1.1

### LINKED ISSUE #171138: beancount-language-server does not build
### Describe the bug
beancount-language-server is currently being built as a node package (https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/node-packages/node-packages.nix#L82268), it is:

- outdated in nixpkgs
- not built and cached by cahce.nixos.org
- fails to build when added to configuration.nix

I believe the reason being it is really a Rust package: https://github.com/polarmutex/beancount-language-server

I tried to create a simple derivation to build it as a Rust package and it seems to be working: https://github.com/bhankas/nixpkgs/commit/b29b62b349475783de9ce183d775a36b8ddc1c43

I would have opened a PR, but since the package is already present under nixpkgs (although in arguably wrong location), the source is same and I am not sure how the case should be handled. Any direction welcome.

If someone wants to cherry-pick my commit and contribute themselves, that is fine as well.

### Steps To Reproduce
1. Add `nodePackages.beancount-langserver` in configuration.nix
2. `nixos-rebuild` boot`

### Expected behavior
`beancount-language-server` should be available in $PATH after nixos-rebuild

### Notify maintainers
Not sure who the maintainer is, as the source is a 4+MB json file.

### PR DESCRIPTION
**Pull request by polarmutex**
###### Description of changes

fixes: #171138

This PR does the following:
-  removes the npm version of the beancount-language-server tool which has been deprecated
- adds the new refactored rust version

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -10098,6 +10098,12 @@
     githubId = 10473184;
     name = "Jia Xiaodong";
   };
+  polarmutex = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:10101]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -31284,6 +31284,8 @@ with pkgs;
 
   beancount = with python3.pkgs; toPythonApplication beancount;
 
+  beancount-language-server = callPackage ../development/tools/beancount-language-server/default.nix {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:31287]:
```suggestion
  beancount-language-server = callPackage ../development/tools/beancount-language-server {};
```

File: pkgs/development/tools/beancount-language-server/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  pname = "beancount-language-server";
+  version = "1.1.1";
+  cargoSha256 = "sha256-NTUs9ADTn+KoE08FikRHrdptZkrUqnjVIlcr8RtDvic=";
+
+  src = fetchFromGitHub {
+    owner = "polarmutex";
+    repo = "beancount-language-server";
+    rev = "v${version}";
+    sha256 = "sha256-CkwNxamkErRo3svJNth2F8NSqlJNX+1S/srKu6Z+mX4=";
+  };
```
Comment (SuperSandro2000) [inline pkgs/development/tools/beancount-language-server/default.nix:16]:
```suggestion
  version = "1.1.1";

  src = fetchFromGitHub {
    owner = "polarmutex";
    repo = "beancount-language-server";
    rev = "v${version}";
    sha256 = "sha256-CkwNxamkErRo3svJNth2F8NSqlJNX+1S/srKu6Z+mX4=";
  };

  cargoSha256 = "sha256-NTUs9ADTn+KoE08FikRHrdptZkrUqnjVIlcr8RtDvic=";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by polarmutex** [comment]
@SuperSandro2000  this is ready for re-review when you get a chance

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 171766` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages marked as broken and skipped:
</summary>
<ul>
<li>teleprompter</li>
<li>timedoctor</li>
</ul>
</details>
<details>

  <summary>1 package failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>cinnamon.xreader: log was empty</li>
  </ul>
</details>
<details>
<summary>
47 packages built:
</summary>
<ul>
<li>balanceofsatoshis</li>
<li>beancount-language-server</li>
<li>bitwarden-cli</li>
<li>castnow</li>
<li>code-server</li>
<li>commitlint</li>
<li>create-cycle-app</li>
<li>deltachat-desktop</li>
<li>epgstation</li>
<li>fast-cli</li>
<li>flexoptix-app</li>
<li>google-clasp</li>
<li>grafana-image-renderer</li>
<li>hyperspace-cli</li>
<li>joplin</li>
<li>morgen</li>
<li>postcss-cli</li>
<li>pyright</li>
<li>redoc-cli</li>
<li>reveal-md</li>
<li>sage</li>
<li>sageWithDoc</li>
<li>shout</li>
<li>slack</li>
<li>teams</li>
<li>theLoungePlugins.plugins.closepms</li>
<li>theLoungePlugins.plugins.giphy</li>
<li>theLoungePlugins.themes.flat-blue</li>
<li>theLoungePlugins.themes.flat-dark</li>
<li>thelounge</li>
<li>uppy-companion</li>
<li>vimPlugins.coc-git</li>
<li>vimPlugins.coc-lists</li>
<li>vimPlugins.coc-metals</li>
<li>vimPlugins.coc-pyright</li>
<li>vimPlugins.coc-stylelint</li>
<li>vscode</li>
<li>vscode-extensions.matklad.rust-analyzer</li>
<li>vscode-extensions.ms-python.vscode-pylance</li>
<li>vscode-extensions.vadimcn.vscode-lldb</li>
<li>vscode-fhs</li>
<li>vscode-with-extensions</li>
<li>vscodium</li>
<li>vscodium-fhs</li>
<li>wasm-text-gen</li>
<li>whalebird</li>
<li>zx</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
balanceofsatoshis:
<details>

got build log for '/nix/store/mi7zw394njgh0q6nmp9qn953n3vlxpyr-balanceofsatoshis-12.8.4' from 'daemon'
When evaluating attribute ‘balanceofsatoshis’:
warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/node-packages/node-env.nix:472:7:

```
    |
472 |       meta = {
    |       ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
warning: name-and-version
Did you mean to pass `pname` instead of `name` to `mkDerivation`?

Near pkgs/development/node-packages/node-env.nix:412:7:

```
    |
412 |       name = "${name}${if version == null then "" else "-${version}"}";
    |       ^
```

Near pkgs/development/node-packages/node-packages.nix:80349:5:

```
      |
80349 |     version = "12.8.4";
      |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/name-and-version.md
warning: build-tools-in-build-inputs
util-linux is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/node-packages/node-env.nix:413:7:

```
    |
413 |       buildInputs = [ tarWrapper python nodejs ]
    |       ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
warning: attribute-typo
A likely typo in the `preRebuild` argument was found, did you mean `postBuild`?

Near pkgs/development/node-packages/node-env.nix:421:14:

```
    |
421 |       inherit dontNpmInstall preRebuild unpackPhase buildPhase;
    |              ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/attribute-typo.md
warning: unused-argument
Unused argument: `nix-gitignore`.
Near pkgs/development/node-packages/node-packages.nix:3:31:

```
  |
3 | {nodeEnv, fetchurl, fetchgit, nix-gitignore, stdenv, lib, globalBuildInputs ? []}:
  |                               ^
```
warning: unused-argument
Unused argument: `stdenv`.
Near pkgs/development/node-packages/node-packages.nix:3:46:

```
  |
3 | {nodeEnv, fetchurl, fetchgit, nix-gitignore, stdenv, lib, globalBuildInputs ? []}:
  |                                              ^
```
warning: unused-argument
Unused argument: `lib`.
Near pkgs/development/node-packages/node-packages.nix:3:54:

```
  |
3 | {nodeEnv, fetchurl, fetchgit, nix-gitignore, stdenv, lib, globalBuildInputs ? []}:
  |                                                      ^
```
</details>
</details>

**Comment by polarmutex** [comment]
@SuperSandro2000 what else do I need to do to get this merged?

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 171766` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>deltachat-desktop</li>
  </ul>
</details>
<details>
  <summary>33 packages built:</summary>
  <ul>
    <li>antennas</li>
    <li>balanceofsatoshis</li>
    <li>beancount-language-server</li>
    <li>bitwarden-cli</li>
    <li>castnow</li>
    <li>commitlint</li>
    <li>create-cycle-app</li>
    <li>epgstation</li>
    <li>gtop</li>
    <li>hueadm</li>
    <li>hyperspace-cli</li>
    <li>imapnotify</li>
    <li>iosevka</li>
    <li>joplin</li>
    <li>mastodon-bot</li>
    <li>node2nix</li>
    <li>pm2</li>
    <li>postcss-cli</li>
    <li>redoc-cli</li>
    <li>reveal-md</li>
    <li>shout</li>
    <li>sloc</li>
    <li>theLoungePlugins.plugins.closepms</li>
    <li>theLoungePlugins.plugins.giphy</li>
    <li>theLoungePlugins.themes.flat-blue</li>
    <li>theLoungePlugins.themes.flat-dark</li>
    <li>thelounge</li>
    <li>triton</li>
    <li>uppy-companion</li>
    <li>vimPlugins.coc-stylelint</li>
    <li>vscode-extensions.matklad.rust-analyzer</li>
    <li>vscode-extensions.vadimcn.vscode-lldb</li>
    <li>zx</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
> @SuperSandro2000 what else do I need to do to get this merged?

Wait for someone to take another look and if everything is looking good it will be merged
